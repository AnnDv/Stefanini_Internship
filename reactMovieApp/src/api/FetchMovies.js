import { CONFIG } from "../config";
import React, { useEffect, useState } from "react";
import MovieCard from "../components/MovieCard";

const API_URL = CONFIG.BASE_URL + CONFIG.POPULAR_MOVIE_URL + CONFIG.API_KEY;

export const FetchMovies = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    fetch(API_URL)
    .then(response => response.json())
    .then(data => { 
      console.log(data.results);
      setMovies(data.results);
    });
  }, [])
  
  return (
    <div className='container'>
      <MovieCard movies={movies}/>    
    </div>
      
  );
}

export default FetchMovies;