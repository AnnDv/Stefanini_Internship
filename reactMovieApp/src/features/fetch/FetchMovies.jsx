import React, { useEffect, useState } from 'react';
import MovieList from '../movieList/MovieList.jsx';
import SearchBox from '../search/SearchMovie.jsx';
import { useGetPopularMoviesQuery, useLazyGetSearchMovieQuery } from '../../store/sliceApi';
// import Debouncer from '../../components/Debounce/Debouncer';

const FetchMovies = () => {
  const { data: movie } = useGetPopularMoviesQuery();
  const [movies, setMovies] = useState([]);
  const [search, setSearch] = useState('');
  const [execute, { data: list }] = useLazyGetSearchMovieQuery();

  // const debouncedSearchTerm = Debouncer(search, 500);

  useEffect(() => {
    if (!search.length) {
      setMovies(movie);
    } else if (search.length >= 3) {
      execute(search);
      setMovies(list);
    }
  }, [execute, search, list, setMovies, movie]);

  return (
    <>
      <nav className='navbar'>
        <h3>Movie TMDB</h3>
          <SearchBox
          search={search}
          setSearch={setSearch}
        />
      </nav>
      <div data-testid="fetch" className='container'>
        <MovieList movies={movies}/>
      </div>
    </>
  );
};

export default FetchMovies;
