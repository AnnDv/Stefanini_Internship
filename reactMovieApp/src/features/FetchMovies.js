import { getData } from "../api/getData";
import { CONFIG } from "../config";
import MovieCard from "../components/MovieCard";
import  React, { useEffect, useState } from "react";

const API_URL = CONFIG.BASE_URL + CONFIG.POPULAR_MOVIE_URL + CONFIG.API_KEY;

const FetchMovies = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    getData(API_URL).then((result) => { 
    setMovies(result)
  })
}, [])

  console.log(movies)

  return (
    <div className='container'>
      <MovieCard movies={movies}/>    
    </div>
  );
}

export default FetchMovies;