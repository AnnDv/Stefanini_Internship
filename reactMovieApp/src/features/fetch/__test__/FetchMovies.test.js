import React from 'react';
import { waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import fetchMock from 'jest-fetch-mock';
import FetchMovies from '../FetchMovies.jsx';
import moviesObject from '../__fixtures__/moviesObject';
import renderWithProviders from '../../../common/customRender.jsx';

describe('FetchMovies', () => {
  it('should render movies after the page was loaded', async () => {
    fetchMock.mockResponseOnce(JSON.stringify(moviesObject));

    const { getByTestId } = renderWithProviders(<FetchMovies/>);

    await waitFor(() => {
      expect(getByTestId('668461')).toBeInTheDocument();
    });
  });

  it('should not render movies after the page was loaded with bad (empty) response', async () => {
    fetchMock.mockResponseOnce(JSON.stringify({}));

    const { getByTestId } = renderWithProviders(<FetchMovies/>);

    await waitFor(() => {
      expect(() => getByTestId('668461')).toThrow();
    });
  });

  it('should not render movies with incorrect id', async () => {
    fetchMock.mockResponseOnce(JSON.stringify(moviesObject));

    const { getByTestId } = renderWithProviders(<FetchMovies/>);

    await waitFor(() => {
      expect(() => getByTestId('66841')).toThrow();
    });
  });
});
