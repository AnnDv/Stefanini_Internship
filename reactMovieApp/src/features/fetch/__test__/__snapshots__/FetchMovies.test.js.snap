// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchMovies should add movie data 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <h3>
        Movie TMDB
      </h3>
      <div
        class="form"
      >
        <input
          class="search_form"
          data-testid="search"
          placeholder="Type to search..."
          type="search"
          value=""
        />
      </div>
      ;
    </nav>
    <div
      class="container"
      data-testid="fetch"
    >
      <div
        class="grid"
        data-testid="list"
      >
        <div
          class="card_description"
          data-testid="m-436270"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/pFlaoHTZeyNkG83vxsAJiGzfSsa.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Black Adam
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-1013860"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/g4yJTzMtOBUTAR2Qnmj8TYIcFVq.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                R.I.P.D. 2: Rise of the Damned
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-829799"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/uGuHHS9SWv7MrFhCH6zoGGd7DA8.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Paradise City
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-988233"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/xFJHb43ZAnnuiDztxZYsmyopweb.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Hex
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-862965"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/iZvzMpREGiqDQ5eYbx8z70qPgst.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Emily the Criminal
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-948276"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/uAeZI1JJbLPq7Bu5dziH7emHeu7.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Lost Bullet 2
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-338958"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/4x3pt6hoLblBeHebUa4OyiVXFiM.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Disenchanted
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-966220"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/lZOODJzwuQo0etJJyBBZJOSdZcW.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Sniper: The White Raven
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-882598"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/aPqcQwu4VGEewPhagWNncDbJ9Xp.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Smile
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-899294"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/5NpXoAi3nEQkEgLO09nmotPfyNa.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Frank and Penelope
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-505642"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/sv1xJUazXeYqALzczSZ3O6nkH75.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Black Panther: Wakanda Forever
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-979924"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/a14BbSHvLgzCdbDD6tAL8OBVKL1.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                On the Line
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-846778"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/uNzgeMetu9l4q9NDw7gtiUFwPOJ.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Margaux
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-724495"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/438QXt1E3WJWb3PqNniK0tAE5c1.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                The Woman King
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-872177"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/9qG7cBhu444BYmcIOeoLJoCBMuO.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Corrective Measures
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-551271"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/4njdAkiBdC5LnFApeXSkFQ78GdT.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Medieval
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-830784"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/irIS5Tn3TXjNi1R9BpWvGAN4CZ1.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Lyle, Lyle, Crocodile
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-1037858"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/sWoYDNPNZs5MtzPbirXV73tIHrA.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                The Soccer Football Movie
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-1018403"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/effgnqJMX2lXvXhKlJt7Pm1TkcU.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Tom and Jerry Snowman's Land
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
        <div
          class="card_description"
          data-testid="m-668461"
        >
          <div>
            <div>
              <img
                alt="movie"
                class="movie_img"
                src="https://image.tmdb.org/t/p/w500/f18kgonrgr8YvEuvshExS4XBGL8.jpg"
              />
            </div>
            <div
              class="title"
            >
              <p>
                Slumberland
              </p>
            </div>
            <div
              class="card-body"
            >
              <button
                class="button"
              >
                View More
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
